internal/builders/maven/README.md

```diff
@@ -86,25 +88,51 @@ jobs:
 
 Now, when you invoke this workflow, the Maven builder will build both your artifacts and the provenance files for them.
 
-You can also release artifacts to Maven Central by adding the following step to your workflow:
+### Releasing directly to Maven Central
+
+You can also release artifacts to Maven Central with [the slsa-github-generator Maven publisher](https://github.com/slsa-framework/slsa-github-generator/blob/main/actions/maven/publish/action.yml) by adding the following step to your workflow:
 
 ```yaml
   publish:
-    needs: build
-    uses: slsa-framework/slsa-github-generator/.github/workflows/publish_maven.yml@v1.7.0
-    with:
-      provenance-download-name: "${{ needs.build.outputs.provenance-download-name }}"
-      provenance-download-sha256: "${{ needs.build.outputs.provenance-download-sha256 }}"
-      target-download-sha256: "${{ needs.build.outputs.target-download-sha256 }}"
-    secrets:
-      maven-username: ${{ secrets.OSSRH_USERNAME }}
-      maven-password: ${{ secrets.OSSRH_PASSWORD }}
-      gpg-key-pass: ${{ secrets.GPG_PASSPHRASE }}
-      gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
+    - name: publish
+      id: publish
+      uses: slsa-framework/slsa-github-generator/actions/maven/publish@v1.7.0
+      with:
+        provenance-download-name: "${{ needs.build.outputs.provenance-download-name }}"
+        provenance-download-sha256: "${{ needs.build.outputs.provenance-download-sha256 }}"
+        target-download-sha256: "${{ needs.build.outputs.target-download-sha256 }}"
+        maven-username: ${{ secrets.OSSRH_USERNAME }}
+        maven-password: ${{ secrets.OSSRH_PASSWORD }}
+        gpg-key-pass: ${{ secrets.GPG_PASSPHRASE }}
+        gpg-private-key: ${{ secrets.GPG_PRIVATE_KEY }}
 ```
 
 Now your workflow will build your artifacts and publish them to a staging repository in Maven Central.
 
+In the above example of the publish Action, the job that invokes the Maven builder is called `build`. The publish Action uses output from that job.
+
+#### Publisher requirements
+
+Besides adding the above workflow to your CI pipeline, you also need to add the following plugin to your `pom.xml`:
+
+```java
+<plugin>
+    <groupId>io.github.adamkorcz</groupId>
+    <artifactId>slsa-hashing-plugin</artifactId>
```

Comment:
how do we publish it? Do we need to create an account? Let me know and I'll start the process.
How do we verify the provenance for this plugin before execution?

Reply:
This is a plugin that I have released to Maven Central under my own account. It currently lives here: https://github.com/AdamKorcz/slsa-hashing-plugin. Ideally we should release this under a slsa-framework account on Maven Central. To do that, we need to:

1. Create a separate repository for it.
2. Create a similar issue as this one: https://issues.sonatype.org/browse/OSSRH-91995
3. Set up the Github repository so we can release the plugin to Maven Central.
4. Release it.

I can do 3 and 4, and you (@laurentsimon) would need to do 1 and 2.

Reply:
quick questions:

1. Can we store the code in this repo rather than in a separate repo?
2. Is it possible to load the plugin from a local .jar (either compile or download) rather than pulling it from Maven central each time? This way we would not need to create an account on Maven central. It also means we can verify its digest before use.
3. Is it possible to use the plugin without asking users to update their pom.xml? How common is it to ask users to update their pom.xml? I'm trying to reduce to a minimum the number of changes required for adoption... but I'm not very familiar with Maven ecosystem :)

/cc @loosebazooka

Reply:
> Can we store the code in this repo rather than in a separate repo?

+1 I'd like to be able to review and apply the same checks and rigor to code in the plugin as we do to the rest of the code.

Reply:
> 1. Can we store the code in this repo rather than in a separate repo?

Yes

> Is it possible to load the plugin from a local .jar (either compile or download) rather than pulling it from Maven central each time?

Yes to both compiling and downloading, although I would argue that the best long-term solution is to release to Maven Central.

> Is it possible to use the plugin without asking users to update their pom.xml? I believe so, yes, by manually downloading it. 

> How common is it to ask users to update their pom.xml? I'm trying to reduce to a minimum the number of changes required for adoption

AFAIK it is the recommended way to consume plugins. See for example https://maven.apache.org/guides/mini/guide-configuring-plugins.html#generic-configuration. As a reference: You need to add ~4 plugins to your `pom.xml` to release to Maven Central.

Reply:
> > 1. Can we store the code in this repo rather than in a separate repo?
> 
> Yes
> 
> > Is it possible to load the plugin from a local .jar (either compile or download) rather than pulling it from Maven central each time?
> 
> Yes to both compiling and downloading, although I would argue that the best long-term solution is to release to Maven Central.

Let's start with a local build then. The problem with maven publication is that we can't verify the package properly yet. Let's do a local compilation for now and we'll use a pre-built jar later (with digest verification).

OK?

Reply:
Added the plugin locally and set up the builder and publisher to install it locally.